jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py

```diff
@@ -0,0 +1,140 @@
+
+    :param normalize: a bool to specify whether to normalize the audio
+     by the sample rate
+    :param dump: a bool to specify whether to dump the segmented audio
+
+    """
+
+    TARGET_SAMPLE_RATE = 16000
+
+    def __init__(
+        self,
+        normalize: bool = False,
+        dump: bool = False,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        # TODO: remove the following temporary fix for torch hub
+        torch.hub._validate_not_a_forked_repo = lambda *args: True
+        self.model, utils = torch.hub.load(
+            repo_or_dir='snakers4/silero-vad', model='silero_vad', force_reload=True
+        )
+        self.model.eval()  # set model to eval mode
+
+        (_, self.get_speech_ts_adaptive, self.save_audio, *_) = utils
+        self.normalize = self._normalize if normalize else lambda audio, _: audio
+        self.dump = dump
+
+    @requests
+    def segment(
+        self,
+        docs: Optional[DocumentArray] = None,
+        parameters: dict = {},
+        **kwargs,
+    ):
+        """
+        Segment all audio docs to chunks using silero-vad
+        :param docs: documents sent to the segmenter. The docs must have either
+        `blob` or `uri` that ends with either `.mp3` or `.wav`.
+        :param parameters: dictionary to override the default parameters.
+        """
+        if not docs:
+            return
+
+        for doc in docs:
+            unnormalized_audio, sample_rate = self._resample(*self._load_raw_audio(doc))
+            audio = self.normalize(unnormalized_audio, sample_rate)
+            speech_timestamps = self.get_speech_ts_adaptive(audio, self.model)
+            doc.blob, doc.tags['sample_rate'] = (
+                audio.cpu().numpy(),
+                sample_rate,
+            )
+
+            if self.dump:
+                dump_dir = Path(self.workspace) / 'audio'
+                dump_dir.mkdir(exist_ok=True)
+                self.save_audio(
+                    str(dump_dir / f'doc_{doc.id}_original.wav'),
+                    unnormalized_audio,
+                    sample_rate,
+                )
+                chunk_path = f'{dump_dir}/chunk_{doc.id}_{{0}}_{{1}}.wav'
+
+            for ts in speech_timestamps:
+                doc.chunks.append(
+                    Document(
+                        blob=audio[ts['start'] : ts['end']].cpu().numpy(),
+                        location=[ts['start'], ts['end']],
+                        tags=doc.tags,
+                    )
+                )
+
+                if self.dump:
+                    self.save_audio(
+                        chunk_path.format(ts['start'], ts['end']),
+                        unnormalized_audio[ts['start'] : ts['end']],
+                        sample_rate,
+                    )
+
+    def _load_raw_audio(self, doc: Document) -> Tuple[torch.Tensor, int]:
+        if doc.blob is not None and doc.tags.get('sample_rate', None) is None:
```

Comment:
```suggestion
        if doc.blob is not None and 'sample_rate' not in doc.tags:
```

Reply:
no, because if ''sample_rate` is None, it will fail. So I think current solution is best

Reply:
Why would someone set `sample_rate` to `None`?

Reply:
As a user, when I see the error "data is blob but sample rate is not provided", I would probably set `sample_rate` to `None` in an attempt to fix the error. In this case, I think it might be useful prevent user from doing this by specifying : `doc.tags.get('sample_rate', None) is None`?